### PR TITLE
Refs #10970 - Rails 4 fixing manifest import errors

### PR DESCRIPTION
**Pull request by johnpmitsch**

### CODE REVIEW

File: app/lib/actions/candlepin/candlepin_listening_service.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def start(suspended_action)
         else
           suspended_action.notify_not_connected("Not Connected")
         end
-      rescue TransportFailure => e
+      rescue => e
```
Comment (johnpmitsch) [inline app/lib/actions/candlepin/candlepin_listening_service.rb]: @jlsherrill I don't any error when I changed to this, I think TransportFailure was being evaluated at some point and throwing uninitialized constant errors. Let me know what you think.

Comment (cpeters) [inline app/lib/actions/candlepin/candlepin_listening_service.rb]: Do we really want to rescue all exceptions here?

Comment (johnpmitsch) [inline app/lib/actions/candlepin/candlepin_listening_service.rb]: @cpeters Probably not, @jlsherrill and I could not find where TransportFailure comes from (not katello, foreman, or qpid), curious to hear if anyone has some insight @thomasmckay do you know anything about this line?

Comment (ehelms) [inline app/lib/actions/candlepin/candlepin_listening_service.rb]: Looks like qpid throws this?

```
/home/vagrant/.rvm/gems/ruby-1.9.3-p448/gems/qpid_messaging-0.30.0/ext/cqpid/cqpid.cpp:    catch(qpid::messaging::TransportFailure& error) {
```

Comment (johnpmitsch) [inline app/lib/actions/candlepin/candlepin_listening_service.rb]: @ehelms oh there it is, thanks. We are seeing alot of issues with uninitialized constants in rails 4, is there a way to reference TransportFailure from a top level?

Comment (ehelms) [inline app/lib/actions/candlepin/candlepin_listening_service.rb]: Did you try `::TransportFailure` ?

Comment (johnpmitsch) [inline app/lib/actions/candlepin/candlepin_listening_service.rb]: @ehelms of course its that simple :) that seems to work just fine, updated the commit.

**Comment by johnpmitsch** [comment]
[test]

**Comment by cpeters** [comment]
ACK

File: app/models/katello/glue/candlepin/product.rb [hunk 2]
```diff
@@ -70,7 +70,7 @@ def orphaned?
       end
 
       def build_product_content(attrs)
-        @productContent = attrs.collect { |pc| Candlepin::ProductContent.new pc }
+        @productContent = attrs.collect { |pc| ::Katello::Candlepin::ProductContent.new pc }
       end
```
Comment (cpeters) [inline app/models/katello/glue/candlepin/product.rb]: This should be...
Katello::Candlepin::ProductContent

Comment (johnpmitsch) [inline app/models/katello/glue/candlepin/product.rb]: `::` at the beginning just means use an absolute path to the module vs. a relative path, my thinking was absolute path would better ensure the correct class is loaded. More info here http://stackoverflow.com/questions/10482772/rubys-double-colon-operator-usage-differences

Comment (komidore64) [inline app/models/katello/glue/candlepin/product.rb]: Though the leadning `::` isn't incorrect, I believe it's unnecessary in this case.

Comment (johnpmitsch) [inline app/models/katello/glue/candlepin/product.rb]: Since either way works, I updated to without leading `::`

**Comment by cpeters** [comment]
@johnpmitsch Can you check the commit again... it looks like rescue => e is still present.

**Comment by johnpmitsch** [comment]
@cpeters not sure what happened there :/ its updated correctly now.

**Comment by cpeters** [comment]
@johnpmitsch wierd.

ACK

**Pull request merged**